### PR TITLE
Data returned in REST api is always strict JSON.

### PR DESCRIPTION
**Pull request by dkeeney**
A slight modification to the REST api.  It now always returns strict JSON so there is no question as to what to expect on the client side.  The client side can decode the JSON before doing anything with the response.

I found that some of my C++ JSON encoding routines were not quite right.  They now wrap all scalar strings in double quotes.  Numbers and keywords 'true', 'false', and 'null' are not quoted.

### CODE REVIEW

**Comment by Lupino** [comment]
I think the error result we should return the follow result.
```
{ "err": "error message" }
```

the right result we return the follow result
```
{ "result": result_json }
```


in py

```
if result.get('err'):
    raise NetworkRESTError(result['err'])

return result['result']
```

File: CommonCompilerConfig.cmake [hunk 1]
```diff
@@ -187,6 +187,7 @@ if(MSVC)
 		BOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS
 		BOOST_NO_WREGEX
 		_SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING
+		_SILENCE_CXX17_RESULT_OF_DEPRECATION_WARNING
```
Comment (breznak) [inline CommonCompilerConfig.cmake:190]: can you revert this and leave it to #846 ?

Comment (dkeeney) [reply CommonCompilerConfig.cmake:190]: Oh, forgot to remove that before submitting....cannot compile locally without it.

**Comment by dkeeney** [comment]
@Lupino  I like your idea a lot of using the {"result": {...}} and {"err": xxxx}

**Comment by dkeeney** [comment]
The ubuntu build hung obtaning Eigen.  This is new.

**Comment by breznak** [comment]
> The ubuntu build hung obtaning Eigen. This is new. 

are we using the gitlab repo, if not I guess we should & could. 
https://github.com/eigenteam/eigen-git-mirror

**Comment by dkeeney** [comment]
Implemented @Lupino's idea of {"result": {...}} and {"err": xxxx}

**Comment by dkeeney** [comment]
Thank you for the review.

**Pull request merged**